cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -358,29 +368,33 @@ private void addTransforms(List<TransformDetail> pipeline,
     public void map(Object key, Object value, Context context) throws IOException, InterruptedException {
       try {
         KeyValue<Object, Object> input = new KeyValue<>(key, value);
-        TransformResponse<Object> recordsAndErrors = transformExecutor.runOneIteration(input);
-        Iterator<Object> transformedRecords = recordsAndErrors.getEmittedRecords();
-        while (transformedRecords.hasNext()) {
-          Object transformedRecord = transformedRecords.next();
-          for (WrappedSink<Object, Object, Object> sink : sinks) {
+        TransformResponse transformResponse = transformExecutor.runOneIteration(input);
+        for (Map.Entry<String, List<Object>> transformedEntry : transformResponse.getSinksResults().entrySet()) {
+          Iterator<Object> transformedRecords = transformedEntry.getValue().iterator();
+          WrappedSink<Object, Object, Object> sink = sinks.get(transformedEntry.getKey());
+          while (transformedRecords.hasNext()) {
```

Comment:
slightly more concise to use `for (Object transformedRecord : transformedEntry.getValue())`